### ISSUE TITLE
Flows not enabled after import

### ISSUE THREAD

**Issue by ManuInNZ** [MicrosoftDocs/power-platform#868]
Hi

Thanks for updating us about the usage of the DeploymentSettingsFile.
However, when using this technique we still seem to have to enable the cloud flows after import.
Am I missing something to ensure the flows are enabled after import?

I.e. we import using a devops service principal which does not own the connections and therefore need to use a PowerApps Admin app to use impersonation and activate the flows.
Since I have a number of flows I trigger their status update in parallel threads 

```Powershell
foreach ($flow in $flows) {
     Start-ThreadJob -Name $flow.name -ThrottleLimit 2000 -ScriptBlock { Set-CrmRecordState -conn $Using:impersonatedconn -EntityLogicalName workflow -Id ($Using:flow).workflowid -StateCode Activated -StatusCode Activated }
}
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ca1df5d8-ef79-c56d-3950-0617da9f976b
* Version Independent ID: ff132857-2fc9-57d0-df24-d42c054cbf1c
* Content: [Pre-populate connection references and environment variables for automated deployments using Power Platform Build Tools - Power Platform](https://docs.microsoft.com/en-us/power-platform/alm/conn-ref-env-variables-build-tools)
* Content Source: [power-platform/alm/conn-ref-env-variables-build-tools.md](https://github.com/MicrosoftDocs/power-platform/blob/main/power-platform/alm/conn-ref-env-variables-build-tools.md)
* Service: **power-platform**
* Sub-service: **alm**
* GitHub Login: @kkanakas
* Microsoft Alias: **kartikka**

**Comment by ManuInNZ** [comment]
No updates eventually ?

Again so far we have no other options that I know of but to loop and activate flows via scripts looping again if there are child flows used.

Cheers
Manu

**Comment by kkanakas** [comment]
Hi @ManuInNZ ,
                       This is a known issue, we are working on addressing this issue in the long run. Just to be clear, you want to activate the flows that you imported via SPN to also be activated via the DevOps Pipeline?

**Comment by ManuInNZ** [comment]
Sorry for the delay, I thought I answered that already but no.
Yes we do deploy/import using the power platform task using a SPN and the deployment settings file.
and yes after import and doing a check of the state, they end up being in Draft mode.
Hence why we have to use an impersonation of the connections owner to enable the flows.
And this is also an issue as he need to have the same owner of the connections otherwise it gets really painful to enable the flows should they contain connections belonging to different users(if even possible).

**Comment by randolphnz** [comment]
Hi ManuInNZ, I just wonder if the impersonation works? From my testing, it's not. Below is how we impersonate:

$impersonatedconn = Get-CrmConnection -ConnectionString $connectionString
$impersonatedconn.OrganizationWebProxyClient.CallerId = $user.CrmRecords[0].systemuserid

**Comment by ManuInNZ** [comment]
> Hi ManuInNZ, I just wonder if the impersonation works? From my testing, it's not. Below is how we impersonate:
> 
> $impersonatedconn = Get-CrmConnection -ConnectionString $connectionString $impersonatedconn.OrganizationWebProxyClient.CallerId = $user.CrmRecords[0].systemuserid

Hi Randolph

At least it used to and may still most of the time. Lately I am not so much involved with this start and stop of flows.
I think about 6 to 8 months ago, the script above stopped to work on some connections and at that time the responsibility of this flows was already transferred to another partner. As far as I can see, the owner of the connections is still required now to start some of the flows.
Alternatively you should consider if not done either to move to using Service Principal based connections or if it is for an actual enterprise-grade usage, move to more suited services such as logics apps or others in the Azure Integration services.
Just my 2cts though.

**Comment by randolphnz** [comment]
Hi Emmanuel,

Thanks for your comments. That script doesn't work for me either. We have
many flows using SP based connections.

However, it doesn't allow me to auto turn on the flows either if we use SP
or service account to execute the pipeline.

It complains 403 errors like "you cannot make this flow in this
environment" when using SP to execute the pipeline.

It complains of permission to connection issues when using a service
account to execute the pipeline as connection is created by another normal
user and shared with SP only.

Hope my explanation makes sense to you.

Regards,
Randy

**Comment by ManuInNZ** [comment]
might be as simple as trying to use a connection not existing indeed in the
environment. is that an application user with enough rights ?
e.g. using a sp from SIT in UAT.
are you you you also have the impersonation rights?

**Comment by randolphnz** [comment]
Yes, both service account and SP are system admin and environment maker.

I am very confused why it still complains that for SP to turn on flies even
though it’s shared with the connections to Dataverse and Gmail in the
flows.

Randy

Emmanuel Auffray ***@***.***>于2023年8月6日 周日下午3:21写道：

> might be as simple as trying to use a connection not existing indeed in
> the
> environment. is that an application user with enough rights ?
> e.g. using a sp from SIT in UAT.
> are you you you also have the impersonation rights?
>
>
